modules/solid_mechanics/src/postprocessors/InteractionIntegral.C

```diff
@@ -154,14 +156,24 @@ InteractionIntegral::computeQpIntegral()
   // Term3 = aux stress * strain * dq_x   (= stress * aux strain * dq_x)
   Real term3 = dq(0,0) * _aux_stress[_qp].doubleContraction(strain_cf);
 
+  // Term4 (thermal strain term) = q * aux_stress * alpha * x1-derivative of theta
+  // - the term including the derivative of alpha is not implemented
+  Real term4 = 0.0;
+  if (_has_temp)
+  {
+    Real aux_stress_trace = _aux_stress[_qp](0,0) + _aux_stress[_qp](1,1) + _aux_stress[_qp](2,2);
+    Real grad_temp_sum = grad_temp_cf(0) + grad_temp_cf(1) + grad_temp_cf(2);
+    term4 = _scalar_q[_qp] * aux_stress_trace * (*_current_instantaneous_thermal_expansion_coef)[_qp] * grad_temp_sum;
```

Comment:
I am not sure if we are looking at the same equation. It seems to me that we should use grad_temp_cf(0) instead of grad_temp_sum. Please correct me if I am wrong :-)